syscall/syscall.csv

```diff
@@ -42,7 +41,7 @@
 "if_indextoname","net/if.h","defined(CONFIG_NETDEV_IFINDEX)","FAR char *","unsigned int","FAR char *"
 "if_nametoindex","net/if.h","defined(CONFIG_NETDEV_IFINDEX)","unsigned int","FAR const char *"
 "insmod","nuttx/module.h","defined(CONFIG_MODULE)","FAR void *","FAR const char *","FAR const char *"
-"ioctl","sys/ioctl.h","!defined(CONFIG_LIBC_IOCTL_VARIADIC)","int","int","int","unsigned long"
+"ioctl","sys/ioctl.h","","int","int","int","..."
```

Comment:
... will not work in system calls.  You must follow the patter of vsyslog:

"nx_vsyslog","nuttx/syslog/syslog.h","","int","int","FAR const IPTR char*","FAR va_list*"

Reply:
It must work, otherwise how open can work with the same impelemntation?

Reply:
Here is how it works:  There is special logic in tools/mksyscall.c:

static bool is_vararg(const char *type, int ndx, int nparms) checks for "..."
The logic at line 310 generates the va_arg() calls.

This is really inefficient.  It could be make a little better using nparms < 6

Reply:
I still don't like the way that that works.  the way that vsyslog works if much cleaner.

This will also assume that every argument is sizeof(uintptr_t).  That could be a problem.  It would be better to pass the va_list all the way to the driver ioctl method and it can properly handle types of differing size.

Reply:
Since all these functions just accept one more argument from standard, we can modify mksyscall not to hard code the max number to 7 instead geting it from syscall.cvs like this:
```
"open","fcntl.h","","int","const char*","int","...1"
```
and change STUB_open to:
```
SYSCALL_LOOKUP(open,                     3, STUB_open)
uintptr_t STUB_open(int nbr, uintptr_t parm1, uintptr_t parm2,
            uintptr_t parm3);
```

Reply:
> Since all these functions just accept one more argument from standard, we can modify mksyscall not to hard code the max number to 7 instead geting it from syscall.cvs like this:
> 
> ```
> "open","fcntl.h","","int","const char*","int","...1"

I think that is a good idea.  That would greatly improve performance.  It doesn't resolve the issue of the type of the argument; it would still assume that it is a uintptr_t even though in reality it is a mode_t or unsigned int.

That would not work for something like syslog() which has an arbitrarily long list of arguments of various types, perhaps as many as 20 or more.

"..." must always be the last argument.  So perhaps "...N" could be followed by a list of N types like: "...1","mode_t" could mean that there is at most one argument of type mode_t.  That would also solve the typing issue.

Reply:
Yes, this is what come up my mind, we can even drop 1 from the list like this:
```
"open","fcntl.h","","int","const char*","int","...","mode_t"
```
The ... just indicate the arguments after ... need get from va_xxx function indirectly.
The support of old form should be removed to avoid the wrong usage in the furture. The function with unlimited number/type arguments must pass the argumnet through va_list* like syslog.